__tests__/commands/_helpers.js

```diff
@@ -113,11 +125,16 @@ export async function run<T, R>(
   if (fixturesLoc) {
     const source = typeof name === 'string' ? name : name.source;
 
-    const dir = path.join(fixturesLoc, source);
-    cwd = await fs.makeTempDir(path.basename(dir));
-    await fs.copy(dir, cwd, reporter);
-    if (typeof name !== 'string') {
-      cwd = path.join(cwd, name.cwd);
+    // if source wasn't set then assume we were given a complete path
+    if (typeof source === 'undefined') {
+      cwd = typeof name !== 'string' ? name.cwd : await fs.makeTempDir();
```

Comment:
These nested conditionals are a bit confusing. Do you think we can find a cleaner way to do this?

Reply:
they are, if I take this one off which is unnecessary then flow complains. I will take a look in another PR see if I can can come up with a clearer logic and put more comments.